test/chai-sql.coffee

```diff
@@ -101,12 +103,21 @@ exports.operandToAbstractSQLFactory = (binds = [], defaultResource = 'pilot', de
 			fieldParts = operand.split('/')
 			if fieldParts.length > 1
 				alias = parentAlias
-				for fieldPart in fieldParts[...-2]
-					alias = "#{alias}.#{fieldPart.replace(/__/g, '-')}"
-				mapping = clientModel.resourceToSQLMappings[fieldParts[fieldParts.length - 2]][fieldParts[fieldParts.length - 1]]
-				mapping = ["#{alias}.#{mapping[0]}", mapping[1...]...]
+				previousResource = _(parentAlias).split('.').last()
+				for resourceName in fieldParts[...-1]
+					sqlName = resourceNameToSqlName(resourceName)
+					sqlNameParts = sqlName.split('-')
+					mapping = _.get(clientModel.relationships[previousResource], sqlNameParts.join('.'))
+					refTable = mapping[1][0]
+					if sqlNameParts.length > 1 and not _.includes(refTable, '-')
+						# Add the verb to tables that don't include the verb already
+						alias = "#{alias}.#{sqlNameParts[0]}-#{refTable}"
+					else
+						alias = "#{alias}.#{refTable}"
+					previousResource = refTable
+				mapping = [alias, _.last(fieldParts)]
```

Comment:
`_.last` gives the last element of the array, which for this array is guaranteed to be a string (as we got the array from splitting a string, and therefore it must be an array of strings with at least one element) - that last element isn't used anywhere else that I can see either (the for loop above specifically iterates over all the elements other than the last).